### PR TITLE
Extract Alerts configuration

### PR DESCRIPTION
**Pull request by fmrsabino**
- The Alerts related configuration and environment keys were extracted to the `AlertsApiModule` and `AlertsControllerModule`.
- By separating the properties of the Alerts feature to its own configuration object which is loaded dynamically, we can make some properties set as required in the application **only if the respective feature is loaded**.
- This effectively reduces the number of configuration properties that is needed to set for the application to launch, when features are under development or a set of features is not needed by the hosting party.
- In order to avoid key collision, the feature level configuration now uses namespaces (with `registerAs`).
  * `alerts-api` for keys used in `/datasources/alerts-api`
  * `alerts-route` for keys used in `/routes/alerts` 

```mermaid
classDiagram
    ConfigurationModule ..> rootConfiguration
    AlertsApiConfigurationModule ..> alertsApiConfiguration
    AlertsConfigurationModule ..> alertsConfiguration
    NestConfigService o-- ConfigurationModule
    NestConfigService o-- AlertsApiConfigurationModule
    NestConfigService o-- AlertsConfigurationModule
    IConfigurationService -- NestConfigService
    FeatureA ..|> IConfigurationService
    FeatureB ..|> IConfigurationService
    FeatureC ..|> IConfigurationService

    <<Interface>> IConfigurationService
    <<Service>> NestConfigService
    <<Module>> ConfigurationModule
    <<Module>> AlertsApiConfigurationModule
    <<Module>> AlertsConfigurationModule

    note for AlertsApiConfigurationModule "Only loaded if feature flag is set to true"
    note for AlertsConfigurationModule "Only loaded if feature flag is set to true"

    class IConfigurationService {
    }
    class NestConfigService {
    }
    class ConfigurationModule {
    }
    class rootConfiguration {
        about
        applicationPort
        ...
    }
    class AlertsApiConfigurationModule {
    }
    class alertsApiConfiguration {
        apiKey
        baseUri
        account
        project
    }
    class AlertsConfigurationModule {
    }
    class alertsConfiguration {
        signingKey
    }

```

### CODE REVIEW

File: src/config/entities/__tests__/configuration.ts [hunk 1]
```diff
@@ -171,7 +164,7 @@ export default (): ReturnType<typeof configuration> => ({
   express: { jsonLimit: '1mb' },
   features: {
     richFragments: true,
-    email: true,
+    email: false,
```
Comment (fmrsabino) [inline src/config/entities/__tests__/configuration.ts:167]: This was set to `false` because otherwise, every test would have to include the now optional Alerts configuration (resulting in more changes).

Since this feature is opt-in, each test should decide if it should include the feature or not.

File: src/datasources/alerts-api/configuration/alerts-api.configuration.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (iamacook) [inline src/datasources/alerts-api/configuration/alerts-api.configuration.ts:1]: Nit: can we add a comment why this is necessary?

File: src/routes/alerts/configuration/alerts.configuration.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (iamacook) [inline src/routes/alerts/configuration/alerts.configuration.ts:1]: Nit: as before regarding a comment.

File: src/datasources/alerts-api/configuration/alerts-api.configuration.module.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      imports: [ConfigModule.forFeature(configFactory)],
+      providers: [
+        { provide: IConfigurationService, useClass: NestConfigurationService },
+      ],
+      exports: [IConfigurationService],
+    };
+  }
+}
+
+export const alertsApiConfigurationModule =
```
Comment (iamacook) [inline src/datasources/alerts-api/configuration/alerts-api.configuration.module.ts:22]: Nit: I was initially confused regarding the naming of this (starting with a lowercase character). What do you think about adjusting to to something like `RegisteredAlertsApiConfigurationModule`?

File: src/datasources/alerts-api/configuration/alerts-api.configuration.ts (hunk 2 above)
Comment (fmrsabino) [reply src/datasources/alerts-api/configuration/alerts-api.configuration.ts]: Not needed. Leftover from local testing.

[3be66bc](https://github.com/safe-global/safe-client-gateway/pull/1372/commits/3be66bcdefeb534bec04fbec96acf7397dca4a5d)

File: src/datasources/alerts-api/configuration/alerts-api.configuration.module.ts (hunk 4 above)
Comment (hectorgomezv) [reply src/datasources/alerts-api/configuration/alerts-api.configuration.module.ts:22]: I'd be in favor of using `SCREAMING_SNAKE_CASE` here 👍

File: src/routes/alerts/alerts.controller.spec.ts [hunk 5]
```diff
@@ -1618,7 +1630,7 @@ describe('Alerts (Unit)', () => {
         const alert = alertBuilder().build();
         const timestamp = Date.now().toString();
         const signature = fakeTenderlySignature({
-          signingKey,
+          signingKey: faker.string.nanoid(32),
```
Comment (hectorgomezv) [inline src/routes/alerts/alerts.controller.spec.ts:1633]: Why is this needed? `signingKey` is already being taken from `src/routes/alerts/configuration/__tests__/alerts.configuration.ts` right?

Comment (fmrsabino) [reply src/routes/alerts/alerts.controller.spec.ts:1633]: It is not because the feature flag for this test is supposed to be false – https://github.com/safe-global/safe-client-gateway/blob/3be66bcdefeb534bec04fbec96acf7397dca4a5d/src/routes/alerts/alerts.controller.spec.ts#L1604

So this test won't have the alerts configuration available.

File: src/datasources/alerts-api/configuration/alerts-api.configuration.module.ts (hunk 4 above)
Comment (iamacook) [reply src/datasources/alerts-api/configuration/alerts-api.configuration.module.ts:22]: I also agree.

**Review by hectorgomezv — APPROVED** [review]
👏🏻
